snap/local/runtime-helpers/bin/security-proxy-post-setup.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -e
+
+# This script is called as a post-stop-command when
+# security-proxy-setup oneshot service stops.
+#
+# Several config options depend on resources that only exist after proxy is 
+# setup. This scripts re-runs the configure hook after the deferred startup 
+# of security-proxy-setup to apply such configurations.
+
+logger "edgex-secretstore-proxy:post-setup: calling configure hook"
+
+# add bin directory to have e.g. secret-config in PATH
+export PATH="$SNAP/bin:$PATH"
+
+$SNAP/snap/hooks/configure
```

Comment:
As discussed on mattermost, I think this is problematic. If this is triggered during installation, it's going to be racing with "seed" change and may cause conflict error ("seed" change in progress...).

The conflict arises because your configure hook code tries to start services with snapctl and since it is triggered "externally" (in this case by systemd) snapd will not consider this to be part of the seeding (install) change.